pkg/api/validate/dynamic/quota.go

```diff
@@ -133,16 +77,18 @@ func (dv *dynamic) ValidateQuota(ctx context.Context, oc *api.OpenShiftCluster)
 	}
 
 	requiredResources := map[string]int{}
-	err := addRequiredResources(requiredResources, oc.Properties.MasterProfile.VMSize, 3)
-	if err != nil {
-		return err
+	vm, ok := vmTypesMap[oc.Properties.MasterProfile.VMSize]
+	if !ok {
+		return fmt.Errorf("unsupported MasterProfile VMSize %s", oc.Properties.MasterProfile.VMSize)
 	}
+	addRequiredResources(requiredResources, vm, 3)
 	//worker node resource calculation
 	for _, w := range oc.Properties.WorkerProfiles {
-		err = addRequiredResources(requiredResources, w.VMSize, w.Count)
-		if err != nil {
-			return err
+		vm, ok := vmTypesMap[w.VMSize]
+		if !ok {
+			return fmt.Errorf("unsupported WorkerProfile VMSize %s", w.VMSize)
 		}
+		addRequiredResources(requiredResources, vm, w.Count)
```

Comment:
Can we encapsulate `vmTypesMap` lookups inside `addRequiredResources`? Basically revert changes to `ValidateQuota` and only modify `addRequiredResources`.

This way we can move instantiation of `vmTypesMap` into `addRequiredResources` (currently at the package level). You raised a good point [here](https://github.com/Azure/ARO-RP/pull/2059#discussion_r850515468) on mutability of `vmTypesMap`. This change will reduce the context of this map and as a result will reduce a chance of accidentally mutating it.

The slight downside is that we will have to have a more generic "unexpected node VMSize" (as currently in master) error instead of error pointing to `WorkerProfile` or `MasterProfile`. But I think it is a minor inconvinience: these errors are not propogated to end customer and SREs will still be easily undderstand the issue. Ideally we should catch it at unit testing time (out of scope of this refactoring) instead of runtime as discussed earlier.